docs/recording/index.md

```diff
@@ -1,5 +1,13 @@
 # Recording with CARET
 
+## Introduction
+
+Recording includes two sub-steps at least; building a target application from source code and executing the target application. After execution, you will find trace data which has events data sampled by tracepoints.
+
+Trace data validation is necessary as the third step because any failure on recording prevents users from analyzing performance correctly. Trace data is not recorded correctly due to any reason if a target application is large and complicated. CARET serves functions to validate trace data for the step.
```

Comment:
How about adding the following text?

```suggestion
Trace data validation is necessary as the third step because any failure on recording prevents users from analyzing performance correctly. Trace data is not recorded correctly due to any reason if a target application is large or if there is an error in the setup procedure. CARET serves functions to validate trace data for the step.
```